vici/session.go

```diff
@@ -173,9 +173,8 @@ func WithSocketPath(path string) SessionOption {
 	})
 }
 
-// withTestConn is a SessionOption used in testing to supply a net.Conn
-// without actually dialing a unix socket.
-func withTestConn(conn net.Conn) SessionOption {
+// WithConn specifies a net.Conn to communicate with charon through.
```

Comment:
I think it would be good to add a comment similar to [this one](https://wiki.strongswan.org/projects/strongswan/wiki/Vici#Protocol-details) explaining that it's recommended to have vici listen on a UNIX socket, but that other protocols may be used so long as you understand and accept any additional risk that may be present.

Also, there are a few places that mention the `net.Conn` field in `sessionOpts` is for testing only. Those should be updated or removed since it's no longer true with this patch :smile:

Reply:
Added a warning similar to the one you linked. Also, I changed the PR to specify a network, address, and `net.Dialer`, so I did not get rid of the `net.Conn` and therefore did not change the comments.